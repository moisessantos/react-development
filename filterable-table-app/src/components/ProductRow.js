import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ProductRow extends Component {
  render() {
    const {name, stocked, price} = this.props.product;

    return (
      <tr>
        <td>
        {
          stocked ? 
            name :
            <span style={{color: 'red'}}>{name}</span>
        }
        </td>
        <td>{price}</td>
      </tr>
    );
  }
}

ProductRow.propTypes = {
  product: PropTypes.shape({
    price: PropTypes.string.isRequired,
    stocked: PropTypes.bool,
    name: PropTypes.string.isRequired,
  }).isRequired,
};
export default ProductRow;
