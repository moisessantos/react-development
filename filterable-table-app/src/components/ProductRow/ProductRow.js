import React, { memo } from 'react';
import PropTypes from 'prop-types';

const ProductRow = ({ product }) => {
    const { name, stocked, price } = product;

    return (
      <tr>
        <td>
        {
          stocked ? 
            name :
            <span style={{color: 'red'}}>{name}</span>
        }
        </td>
        <td>{price}</td>
      </tr>
    );
  };

ProductRow.propTypes = {
  product: PropTypes.shape({
    price: PropTypes.string.isRequired,
    stocked: PropTypes.bool,
    name: PropTypes.string.isRequired,
  }).isRequired,
};

ProductRow.displayName = 'ProductRow';

export default memo(ProductRow);
