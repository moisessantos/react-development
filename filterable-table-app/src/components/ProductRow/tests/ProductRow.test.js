import React from 'react';
import { shallow } from 'enzyme';
import ProductRow from '../';

describe('ProductRow component', () => {
  const product = {
    price: '100 â‚¬',
    name: 'Ball',
    stocked: true
  }

  it('should render snapshot', () => {
    const wrapper = shallow(<ProductRow product = {product} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should add style when isStocked is false', () => {
    const wrapper = shallow(<ProductRow product = {{ ...product, stocked: false } } />);
    
    const style = wrapper.find('span').get(0).props.style;
    expect(style.color).toBe('red');
  });
});
