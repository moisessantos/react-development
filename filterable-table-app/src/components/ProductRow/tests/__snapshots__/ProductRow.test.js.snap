// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRow component should render snapshot 1`] = `
<tr>
  <td>
    Ball
  </td>
  <td>
    100 â‚¬
  </td>
</tr>
`;
